python/cudf/cudf/_lib/csv.pyx

```diff
@@ -429,6 +431,25 @@ def read_csv(
         column_names=meta_names
     ))
 
+    if dtype is not None:
+        if isinstance(dtype, abc.Mapping):
```

Comment:
Is it possible for this to be a generic `Mapping`, or is it guaranteed to be a `dict`? If you only care about `dict`s and not `dict`-like objects, then avoid the abstract class check in favor of `isinstance(dtype, dict)`. It can be considerably faster.

Reply:
I suppose this might come from unsanitized user input with random dict-like classes as the `dtype`...? Maybe this is the best we can do (here and in the other thread below).

Reply:
Our check's aren't much tighter checking for `dict` because in pandas `dtype` is super overloaded param, where for one example they support `defaultdict` too: https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html